packages/playground/src/index.ts

```diff
@@ -0,0 +1,283 @@
+            if (received.data.interaction == interactionObject.name) {
+              console.log("Running interaction: " + interactionObject.name);
+
+              // run the interaction function and store the related run info
+              asyncLocalStorage.run(new Map(), async () => {
+                const store = asyncLocalStorage.getStore() as any;
+
+                store.set("id", received.data.id); // interaction run ID
+                store.set("stepOverrides", received.data.stepOverrides);
+
+                let runOutput;
+                if (received.data.inputs) {
+                  // use inputs in the received message
+                  runOutput = await interactionObject.interaction(
+                    received.data.inputs,
+                  );
+                } else {
+                  // use defaults in the interaction object
+                  runOutput = await interactionObject.interaction(
+                    interactionObject.inputFields,
+                  );
+                }
+
+                // collect data stored within getStepInfo function
+                const stepsArray = [
+                  {
+                    name: store.get("stepName"),
+                    inputs: store.get("stepInputs"),
+                  },
+                ];
+
+                // create a new runResponse event
+                const runResponseEvent = {
+                  id: uuidv4(),
+                  for: received.from,
+                  data: {
+                    type: "runResponse",
+                    id: store.get("id"), // this is the ID for the run
+                    outputs: runOutput,
+                    steps: stepsArray,
+                  },
+                };
+
+                console.log(
+                  "-> Sending event of type " + runResponseEvent.data.type,
+                );
+                console.log(runResponseEvent);
+                console.log(JSON.stringify(runResponseEvent));
+
+                ws.send(JSON.stringify(runResponseEvent));
+              });
+            }
+          }
+        }
+      }
+    });
+
+    ws.addEventListener("error", (e) => {
+      console.error("error", e);
+    });
+
+    ws.addEventListener("close", () => {
+      console.error("close");
+    });
+  }
+
+  public registerCustomType(typeName: string) {
+    this.registeredCustomTypes.push(typeName);
+    //console.log("registeredCustomTypes: " + this.registeredCustomTypes);
+    return { output: typeName };
+  }
+
+  public registerCustomObject(
+    typeName: string,
+    objectName: string,
+    object: object,
+  ) {
+    const customObject = {
+      typeName: typeName,
+      objectName: objectName,
+      object: object,
+    };
+    this.registeredCustomObjects.push(customObject);
+    /*
+    console.log(
+      "registeredCustomObjects: " + JSON.stringify(this.registeredCustomObjects)
+    );
+    */
+  }
+
+  public registerInteraction(
```

Comment:
```
public registerInteraction<FieldType>(
  name: string,
  inputFields: FieldType,
  outputFields: object,
  interaction: InteractionFunc<{ [K in keyof FieldType]: any }>,
) {
  ... 
}
```

Can we do more clever typing here to improve DX similar to the above (not sure if that actually works)? No worries for the first version, but definitely for GA, we'll need to refine these types to be more user friendly.

At the very least, we shouldn't `any` type the interaction function.

Reply:
https://github.com/gentrace/gentrace-node/pull/384